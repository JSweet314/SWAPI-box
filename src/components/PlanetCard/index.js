import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const PlanetCard = ({card, favorites, handleOnClick}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  const confirmResidents = () => {
    if (!card.residents.length) {
      return 'n/a';
    }
    return card.residents.join(', ');
  };

  const addCommas = (number) => {
    return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
  };
  
  const {name, population, terrain, climate} = card;
 
  const residents = confirmResidents();
   
  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{name}</h3>
          <button 
            onClick={() => handleOnClick(card, 'planets')}
            className={selected}></button>
        </div>
        <p>Population: {addCommas(population)}</p>
        <p>Terrain: {terrain}</p>
        <p>Climate: {climate}</p>
        <p>Residents: {residents}</p>
      </div>
    </article>
  );  
};

PlanetCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired,
    terrain: PropTypes.string.isRequired,
    climate: PropTypes.string.isRequired,
    residents: PropTypes.array.isRequired
  }).isRequired,
  favorites: PropTypes.array.isRequired,
  handleOnClick: PropTypes.func.isRequired
};

export default PlanetCard;