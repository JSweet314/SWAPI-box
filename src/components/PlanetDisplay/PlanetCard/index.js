import React from 'react';
import './style.css';
import PropTypes from 'prop-types';

const PlanetCard = ({card, favorites}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  const confirmResidents = () => {
    if (!card.residents.length) {
      return 'n/a';
    }
    return card.residents.join(', ');
  };
  
  const {name, population, terrain, climate} = card;
  const residents = confirmResidents();

    
  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{name}</h3>
          <button className={selected}></button>
        </div>
        <p>Population: {population}</p>
        <p>Terrain: {terrain}</p>
        <p>Climate: {climate}</p>
        <p>Residents: {residents}</p>
      </div>
    </article>
  );  
};

PlanetCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired,
    terrain: PropTypes.string.isRequired,
    climate: PropTypes.string.isRequired,
    residents: PropTypes.array.isRequired
  }).isRequired
};

export default PlanetCard;