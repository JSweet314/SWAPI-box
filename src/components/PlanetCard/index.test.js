import React from 'react';
import {shallow} from 'enzyme';
import PlanetCard from './index';

describe('PlanetCard', () => {
  /* eslint-disable no-undef */
  const mockHandleOnClick = jest.fn();
  /* eslint-enable no-undef */
  let card;

  beforeEach(() => {
    card = {
      name: 'tatooine',
      climate: 'arrid',
      terrain: 'mountains',
      population: '1000',
      residents: ['Tom', 'Dick', 'Harry']
    };
  });

  it('should match a snapshot', () => {
    const wrapper = shallow(
      <PlanetCard
        handleOnClick={mockHandleOnClick}
        favorites={[]}
        card={card} />,
      { disableLifecycleMethods: true }
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should show as favorited if name matches obj in favorites array', () => {
    const wrapper = shallow(
      <PlanetCard
        handleOnClick={mockHandleOnClick}
        favorites={[{ name: 'tatooine' }]}
        card={card} />,
      { disableLifecycleMethods: true }
    );

    expect(wrapper.find('.selected').length).toEqual(1);
  });

  it('should render "n/a" for planets w/ no residents', () => {
    card.residents = [];
    const wrapper = shallow(
      <PlanetCard
        handleOnClick={mockHandleOnClick}
        favorites={[{ name: 'tatooine' }]}
        card={card} />,
      { disableLifecycleMethods: true }
    );

    expect(wrapper).toMatchSnapshot();
  });
});