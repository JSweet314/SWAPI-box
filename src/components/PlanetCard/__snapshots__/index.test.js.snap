// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    card={
        Object {
            "climate": "arrid",
            "name": "tatooine",
            "population": "1000",
            "residents": Array [
              "Tom",
              "Dick",
              "Harry",
            ],
            "terrain": "mountains",
          }
    }
    favorites={Array []}
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="info-card__top-line"
        >
                <h3>
                        tatooine
                </h3>
                <button
                        className=""
                        onClick={[Function]}
                />
        </div>
        <p>
                Population: 
                1,000
        </p>
        <p>
                Terrain: 
                mountains
        </p>
        <p>
                Climate: 
                arrid
        </p>
        <p>
                Residents: 
                Tom, Dick, Harry
        </p>
</div>,
      "className": "info-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="info-card__top-line"
>
            <h3>
                        tatooine
            </h3>
            <button
                        className=""
                        onClick={[Function]}
            />
</div>,
          <p>
            Population: 
            1,000
</p>,
          <p>
            Terrain: 
            mountains
</p>,
          <p>
            Climate: 
            arrid
</p>,
          <p>
            Residents: 
            Tom, Dick, Harry
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                tatooine
</h3>,
              <button
                className=""
                onClick={[Function]}
/>,
            ],
            "className": "info-card__top-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "tatooine",
              },
              "ref": null,
              "rendered": "tatooine",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              "1,000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            "1,000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              "mountains",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            "mountains",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              "arrid",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            "arrid",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              "Tom, Dick, Harry",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            "Tom, Dick, Harry",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="info-card__top-line"
          >
                    <h3>
                              tatooine
                    </h3>
                    <button
                              className=""
                              onClick={[Function]}
                    />
          </div>
          <p>
                    Population: 
                    1,000
          </p>
          <p>
                    Terrain: 
                    mountains
          </p>
          <p>
                    Climate: 
                    arrid
          </p>
          <p>
                    Residents: 
                    Tom, Dick, Harry
          </p>
</div>,
        "className": "info-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info-card__top-line"
>
              <h3>
                            tatooine
              </h3>
              <button
                            className=""
                            onClick={[Function]}
              />
</div>,
            <p>
              Population: 
              1,000
</p>,
            <p>
              Terrain: 
              mountains
</p>,
            <p>
              Climate: 
              arrid
</p>,
            <p>
              Residents: 
              Tom, Dick, Harry
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  tatooine
</h3>,
                <button
                  className=""
                  onClick={[Function]}
/>,
              ],
              "className": "info-card__top-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "tatooine",
                },
                "ref": null,
                "rendered": "tatooine",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Population: ",
                "1,000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Population: ",
              "1,000",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Terrain: ",
                "mountains",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Terrain: ",
              "mountains",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Climate: ",
                "arrid",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Climate: ",
              "arrid",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Residents: ",
                "Tom, Dick, Harry",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Residents: ",
              "Tom, Dick, Harry",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PlanetCard should render "n/a" for planets w/ no residents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    card={
        Object {
            "climate": "arrid",
            "name": "tatooine",
            "population": "1000",
            "residents": Array [],
            "terrain": "mountains",
          }
    }
    favorites={
        Array [
            Object {
              "name": "tatooine",
            },
          ]
    }
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="info-card__top-line"
        >
                <h3>
                        tatooine
                </h3>
                <button
                        className="selected"
                        onClick={[Function]}
                />
        </div>
        <p>
                Population: 
                1,000
        </p>
        <p>
                Terrain: 
                mountains
        </p>
        <p>
                Climate: 
                arrid
        </p>
        <p>
                Residents: 
                n/a
        </p>
</div>,
      "className": "info-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="info-card__top-line"
>
            <h3>
                        tatooine
            </h3>
            <button
                        className="selected"
                        onClick={[Function]}
            />
</div>,
          <p>
            Population: 
            1,000
</p>,
          <p>
            Terrain: 
            mountains
</p>,
          <p>
            Climate: 
            arrid
</p>,
          <p>
            Residents: 
            n/a
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                tatooine
</h3>,
              <button
                className="selected"
                onClick={[Function]}
/>,
            ],
            "className": "info-card__top-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "tatooine",
              },
              "ref": null,
              "rendered": "tatooine",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "selected",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              "1,000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
            "1,000",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain: ",
              "mountains",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Terrain: ",
            "mountains",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate: ",
              "arrid",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Climate: ",
            "arrid",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents: ",
              "n/a",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Residents: ",
            "n/a",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="info-card__top-line"
          >
                    <h3>
                              tatooine
                    </h3>
                    <button
                              className="selected"
                              onClick={[Function]}
                    />
          </div>
          <p>
                    Population: 
                    1,000
          </p>
          <p>
                    Terrain: 
                    mountains
          </p>
          <p>
                    Climate: 
                    arrid
          </p>
          <p>
                    Residents: 
                    n/a
          </p>
</div>,
        "className": "info-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info-card__top-line"
>
              <h3>
                            tatooine
              </h3>
              <button
                            className="selected"
                            onClick={[Function]}
              />
</div>,
            <p>
              Population: 
              1,000
</p>,
            <p>
              Terrain: 
              mountains
</p>,
            <p>
              Climate: 
              arrid
</p>,
            <p>
              Residents: 
              n/a
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  tatooine
</h3>,
                <button
                  className="selected"
                  onClick={[Function]}
/>,
              ],
              "className": "info-card__top-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "tatooine",
                },
                "ref": null,
                "rendered": "tatooine",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "selected",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Population: ",
                "1,000",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Population: ",
              "1,000",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Terrain: ",
                "mountains",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Terrain: ",
              "mountains",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Climate: ",
                "arrid",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Climate: ",
              "arrid",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Residents: ",
                "n/a",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Residents: ",
              "n/a",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
